### PR TITLE
Support boxed primitives in @StateVar

### PR DESCRIPTION
**Pull request by jsyrjala**
Add support for boxed primitives (Integer, Float etc) to @StateVar.

Initialize static variables in static initialization block in WorkflowDefinitionScanner.

### CODE REVIEW

File: nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java [hunk 1]
```diff
@@ -38,9 +33,20 @@
 
   private static final Logger logger = getLogger(WorkflowDefinitionScanner.class);
 
-  private static final Set<Type> knownImmutableTypes = new LinkedHashSet<>();
-  {
-    knownImmutableTypes.addAll(asList(Boolean.TYPE, Boolean.class, Byte.TYPE, Byte.class, Character.TYPE, Character.class, Short.TYPE, Short.class, Integer.TYPE, Integer.class, Long.TYPE, Long.class, Float.TYPE, Float.class, Double.TYPE, Double.class, String.class, BigDecimal.class, BigInteger.class, Enum.class));
+  private static final Set<Object> boxedPrimitiveTypes;
```
Comment (efonsell) [inline nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java]:
```suggestion
  private static final Set<Object> boxedPrimitiveTypes = Stream
    .of(Boolean.class, Byte.class, Integer.class, Long.class, Float.class, Double.class)
    .collect(collectingAndThen(toCollection(LinkedHashSet::new), Collections::unmodifiableSet));
```

File: nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java [hunk 2]
```diff
@@ -38,9 +33,20 @@
...
+  static {
+    Set<Object> set = new LinkedHashSet<>();
+    set.addAll(asList(Boolean.class, Byte.class, Integer.class, Long.class, Float.class, Double.class));
+    boxedPrimitiveTypes = Collections.unmodifiableSet(set);
+  }
+
+  private static final Set<Type> knownImmutableTypes;
```
Comment (efonsell) [inline nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java]:
```suggestion
  private static final Set<Type> knownImmutableTypes = Stream
    .of(Boolean.TYPE, Boolean.class, Byte.TYPE, Byte.class, Character.TYPE, Character.class, Short.TYPE, Short.class,
      Integer.TYPE, Integer.class, Long.TYPE, Long.class, Float.TYPE, Float.class, Double.TYPE, Double.class, String.class,
      BigDecimal.class, BigInteger.class, Enum.class)
    .collect(collectingAndThen(toCollection(LinkedHashSet::new), Collections::unmodifiableSet));
```

File: nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java (hunk 1 above)
Comment (gmokki) [reply nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java]: Should the type be `Set<Class<?>>`

File: nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java (hunk 2 above)
Comment (jsyrjala) [reply nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java]: Fixed

File: nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java (hunk 1 above)
Comment (jsyrjala) [reply nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java]: fixed

File: nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java [hunk 3]
```diff
@@ -17,12 +19,8 @@
 import java.lang.reflect.Type;
 import java.math.BigDecimal;
 import java.math.BigInteger;
-import java.util.ArrayList;
-import java.util.LinkedHashMap;
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (efonsell) [inline nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java]: Expand this?

Comment (jsyrjala) [reply nflow-engine/src/main/java/io/nflow/engine/internal/workflow/WorkflowDefinitionScanner.java]: Expanded

**Comment by efonsell** [comment]
Update CHANGELOG.md

**Pull request merged**